docs/kmigrator.md

```diff
@@ -0,0 +1,168 @@
+ - `models.SET_NULL` -- set the reference to NULL (requires the field to be nullable). For instance, when you delete a User, you might want to keep the comments he posted on blog posts, but say it was posted by an anonymous (or deleted) user. SQL equivalent: SET NULL.
+ - `models.CASCADE` -- when the referenced object is deleted, also delete the objects that have references to it (when you remove a blog post for instance, you might want to delete comments as well). SQL equivalent: CASCADE.
+
+## kmigrator table options
+
+you can set options: 
+ - `constraints` -- to create database level constrains
+ - `indexes` -- to create database level indexes
+
+#### constraints
+
+Example:
+
+```
+const User = new GQLListSchema('User', {
+    schemaDoc: 'Individual / person who have some access to service',
+    fields: {
+        type: {
+            schemaDoc: 'We have b2b and b2c user segments',
+            type: Select,
+            dataType: 'enum',
+            options: ['b2b', 'b2c', 'sys'],
+            defaultValue: 'b2b',
+            isRequired: true,
+        },
+        phone: {
+            schemaDoc: 'Phone. In international E.164 format without spaces',
+            type: Text,
+            isRequired: true,
+        },
+        ...
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['type', 'phone'],
+                name: 'unique_type_and_phone',
+            },
+        ],
+    },
+})
+```
+
+This example generate SQL like so: `ALTER TABLE "User" ADD CONSTRAINT "unique_type_and_phone" UNIQUE ("type", "phone");`
+
+Another example:
+```
+const User = new GQLListSchema('User', {
+    schemaDoc: 'Individual / person who have some access to service',
+    fields: {
+        phone: {
+            schemaDoc: 'Phone. In international E.164 format without spaces',
+            type: Text,
+            isRequired: false,
+        },
+        email: {
+            schemaDoc: 'Email',
+            type: Text,
+            isRequired: false,
+        },
+        ...
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.CheckConstraint',
+                check: 'Q(phone__isnull=False) | Q(email__isnull=False)',
+                name: 'has_phone_or_email',
+            },
+        ],
+    },
+})
+```
+
+This example generate SQL like so: `ALTER TABLE "User" ADD CONSTRAINT "has_phone_or_email" CHECK (("phone" IS NOT NULL OR "email" IS NOT NULL));`
+
+You can also define a `condition` for `models.UniqueConstraint`:
+
+```
+{
+    type: 'models.UniqueConstraint',
+    fields: ['type', 'phone'],
+    condition: 'Q(deletedAt__isnull=True)',
+    name: 'unique_type_and_phone',
+},
+```
+
+SQL result: `CREATE UNIQUE INDEX "unique_type_and_phone" ON "User" ("type", "phone") WHERE "deletedAt" IS NULL;`
+
+Another example:
+```
+{
+    type: 'models.CheckConstraint',
```

Comment:
Can we simplify interface for developer and not expose internal kmigrator conventions?

```suggestion
    type: 'check',
```